common/HttpServer.js

```diff
@@ -67,6 +67,12 @@ class HttpServer {
       process.exit(2);
     }, 500);
   }
+
+  static immediateShutdown() {
+    logger.info('App shutting down now.');
+    pubsub.publish(CONST.TOPIC.APP_SHUTTING_DOWN);
```

Comment:
Isn't `pubsub.publish` asynchronous in nature? If so, this has virtually no effect because the process exits immediately next